File: parser/lsp_model.py
class Position(object):
    def __init__(self, line: int, character: int):
        self.__line = line
        self.__character = character

    @property
    def line(self):
        return self.__line

    @property
    def character(self):
        return self.__character

    def __str__(self):
        return " { line: " + str(self.__line) + ", character: " + str(self.__character) + " }"


class Range(object):
    def __init__(self, start: Position, end: Position):
        self.__start = start
        self.__end = end

    @property
    def start(self):
        return self.__start

    @property
    def end(self):
        return self.__end

    def __str__(self):
        return " { start: " + str(self.__start) + " end: " + str(self.__end) + " }"

class Location(object):
    def __init__(self, uri: str, range: Range):
        self.__uri = uri
        self.__range = range


    @property
    def uri(self):
        return self.__uri

    @property
    def range(self):
        return self.__range

    def __str__(self):
        return " { uri:" + self.__uri + ", range: " + str(self.__range) + " }"

File: go-lsp/src/lsp_model.go
package main

type MethodName string

type LSPService struct{}

const (
	StopCharacter = "\r\n\r\n"
)

const (
	FIND_REFERENCES  MethodName = "textDocument/references"
	CHANGED_DOCUMENT MethodName = "textDocument/didChange"
	INITIALIZE       MethodName = "initialize"
	INITIALIZED      MethodName = "initialized"
)

type TextDocumentKind int

const (
	NONE        TextDocumentKind = 0
	FULL        TextDocumentKind = 1
	INCREMENTAL TextDocumentKind = 2
)

type TextDocumentIdentifier struct {
	Uri string `json:"uri"`
}

type RPCRequest struct {
	JSONRPC string      `json:"jsonrpc"`
	Method  string      `json:"method"`
	Params  interface{} `json:"params,omitempty"`
	ID      int         `json:"id"`
}

type RPCNotification struct {
	JSONRPC string      `json:"jsonrpc"`
	Method  string      `json:"method"`
	Params  interface{} `json:"params,omitempty"`
}

type RPCResponse struct {
	JSONRPC string      `json:"jsonrpc"`
	Result  interface{} `json:"result,omitempty"`
	ID      int         `json:"id"`
}

type RPCError struct {
	Code    int         `json:"code"`
	Message string      `json:"message"`
	Data    interface{} `json:"data"`
}

//REFERENCES
type Position struct {
	Line      float64 `json:"line"`
	Character float64 `json:"character"`
}

type Range struct {
	Start Position `json:"start"`
	End   Position `json:"end"`
}

type Location struct {
	Uri   string `json:"uri"`
	Range Range  `json:"range"`
}

type ReferenceParams struct {
	TextDocument TextDocumentIdentifier `json:"textDocument"`
	Position     Position               `json:"position"`
}

type ReferenceResponse struct {
	Locations []Location `json:"locations"`
}

type ContentChanges struct {
	Text string `json:"text"`
}

type DidChangeTextDocumentParams struct {
	TextDocument   TextDocumentIdentifier `json:"textDocument"`
	ContentChanges ContentChanges         `json:"contentChanges"`
	Version        int                    `json:"version"`
}

type InitializeResult struct {
	Capabilities ServerCapabilities `json:"capabilities,omitempty"`
}

type ServerCapabilities struct {
	TextDocumentSync  TextDocumentKind `json:"textDocumentSync"`
	ReferenceProvider bool             `json:"referencesProvider"`
}
